src/bin/add/args.rs

```diff
@@ -182,6 +194,12 @@ impl Args {
                 dependency = dep.set_version(&v);
             }
 
+            // Set the registry after getting the latest version as
+            // get_latest_dependency returns a registry-less Dependency
```

Comment:
It would complicate the code a bit more so I'll leave as is. (`get_latest_dependency` accepts a registry URL which both cargo-upgrade and cargo-add can easily provide. It's nontrivial for cargo-upgrade to provide a registry name.)